### ISSUE TITLE
Research test automation

### ISSUE THREAD

**Issue by Elyniss** [WalletConnect/WalletConnectKotlinV2#176]

**Comment by Elyniss** [comment]
Research branch: [research/automation_tests](https://github.com/WalletConnect/WalletConnectKotlinV2/tree/research/automation_tests)

TLDR; Automation on demand is achievable right now. However, integrating it into Github action to be available on pull requests/push requires running the action on a self-hosted runner with an M1 chip which is currently missing.

What was accomplished: 
I created sample e2e tests that could open our sample apps and chrome browser and then create a WC session between Dapp integrating JS SDK and Wallet integrating Kotlin SDK. 

How was it accomplished:
A self-hosted runner must run [Appium](https://appium.io/) server which will send commands to an emulated device on the same self-hosted runner to perform various UI actions across Chrome browser, Kotlin Wallet and Kotlin Dapp.

Benefits:
 - This enables us to run all necessary test cases to ensure Kotlin SDK is properly implemented and communicates appropriately with JS SDK. 
 - Quick (I estimate withself-hostedd runners withan  M1 chip the pipeline would take a couple of minutes)

Drawbacks:
M1 self-hosted runner for Github Actions is required. Currently, there is no easy/cheap option to rent such a host. I only found one provider [Scaleway](https://www.scaleway.com/en/docs/compute/apple-silicon/quickstart/) where renting an Apple Mac mini hardware (fifth generation) is possible for a cost of `2,40 €` per day. M1 Mac Mini could be also provided by ourselves.

Possible other approaches:
Another more lengthy approach would be using something like [BrowserStack](https://www.browserstack.com/). We would then build .apks for Kotlin Wallet and Kotlin Dapp with provided Github Actions runners, then upload them to BrowserStack, where we could run our tests. I believe it would take much more time to verify since mere building the .apk would take around 10 minutes of pipeline computation.
